File: task-03/ruby/subtask-3.rb
print "Enter the number: "
n = gets.to_i

for i in 1..n
  for j in 1..(n-i)
    print "  "
  end
  for k in 1..i
    print "* "
  end
  for l in 1..(i-1)
    print "* "
  end
  puts
end

for i in (n-1).downto(1)
  for j in n.downto(i+1)
    print "  "
  end
  for k in 1..i
    print "* "
  end
  for l in 1..(i-1)
    print "* "
  end
  puts
end

File: task-03/cpp/subtask-3.cpp
#include <iostream>

int main() {
    int n;

    std::cout << "Enter the number: ";
    std::cin >> n;

    for (int i = 1; i <= n; i++) {
        for (int j = 0; j < n - i; j++) {
            std::cout << " ";
        }
        int k; 
        for (k = 0; k < i; k++) {
            std::cout << "*";
        }
        for (int l = 0; l < k - 1; l++) {
            std::cout << "*";
        }
        std::cout << std::endl;
    }

    for (int i = n - 1; i >= 1; i--) {
        for (int j = 0; j < n - i; j++) {
            std::cout << " ";
        }
        int k; 
        for (k = 0; k < i; k++) {
            std::cout << "*";
        }
        for (int l = 0; l < k - 1; l++) {
            std::cout << "*";
        }
        std::cout << std::endl;
    }

    return 0;
}
